File: test-suite-docs-java/src/test/java/io/micronaut/testresources/docsupport/DocsTestResourcesClientInjector.java
package io.micronaut.testresources.docsupport;

import io.micronaut.test.extensions.testresources.TestResourcesClientHolder;
import org.junit.platform.launcher.TestExecutionListener;
import org.junit.platform.launcher.TestPlan;

/**
 * Installs the {@link DocsTestResourcesClient} for the duration of the test plan.
 */
public class DocsTestResourcesClientInjector implements TestExecutionListener {
    @Override
    public void testPlanExecutionStarted(TestPlan testPlan) {
        TestResourcesClientHolder.set(new DocsTestResourcesClient());
    }

    @Override
    public void testPlanExecutionFinished(TestPlan testPlan) {
        TestResourcesClientHolder.set(null);
    }
}

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/testresources/docsupport/DocsTestResourcesClientInjector.kt
package io.micronaut.testresources.docsupport

import io.micronaut.test.extensions.testresources.TestResourcesClientHolder
import org.junit.platform.launcher.TestExecutionListener
import org.junit.platform.launcher.TestPlan

/**
 * Installs the [DocsTestResourcesClient] for the duration of the test plan.
 */
class DocsTestResourcesClientInjector : TestExecutionListener {
    override fun testPlanExecutionStarted(testPlan: TestPlan) {
        TestResourcesClientHolder.set(DocsTestResourcesClient())
    }

    override fun testPlanExecutionFinished(testPlan: TestPlan) {
        TestResourcesClientHolder.set(null)
    }
}
